test/e2e/suite/scenario/crl.go

```diff
@@ -0,0 +1,316 @@
+				).
+				NoMatchErrKeyWords(
+					"is revoked",
+				)
+
+			// verify with cache
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-d").
+				MatchKeyWords(
+					VerifySuccessfully,
+				).
+				MatchErrKeyWords(
+					"Retrieving crl bundle from file cache with key",
+					"OCSP check failed with unknown error and fallback to CRL check for certificate #2",
+				).
+				NoMatchErrKeyWords(
+					"CRL file cache miss",
+					"Storing crl bundle to file cache with key",
+					"is revoked",
+				)
+		})
+	})
+
+	It("failed with revoked leaf certificate", func() {
+		Host(CRLOptions(), func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost) {
+			notation.Exec("sign", artifact.ReferenceWithDigest()).
+				MatchKeyWords(SignSuccessfully)
+
+			utils.LeafCRLRevoke()
+			utils.IntermediateCRLUnrevoke()
+
+			// verify without cache
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-d").
+				MatchErrKeyWords(
+					VerifyFailed,
+					"CRL file cache miss",
+					"Retrieving crl bundle from file cache with key",
+					"Storing crl bundle to file cache with key",
+					"is revoked",
+					"OCSP check failed with unknown error and fallback to CRL check for certificate #2",
+				)
+
+			// verify with cache
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-d").
+				MatchErrKeyWords(
+					VerifyFailed,
+					"Retrieving crl bundle from file cache with key",
+					"is revoked",
+					"OCSP check failed with unknown error and fallback to CRL check for certificate #2",
+				).
+				NoMatchErrKeyWords(
+					"CRL file cache miss",
+					"Storing crl bundle to file cache with key",
+				)
+		})
+	})
+
+	It("failed with revoked intermediate certificate", func() {
+		Host(CRLOptions(), func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost) {
+			notation.Exec("sign", artifact.ReferenceWithDigest()).
+				MatchKeyWords(SignSuccessfully)
+
+			utils.LeafCRLUnrevoke()
+			utils.IntermediateCRLRevoke()
+
+			// verify without cache
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-d").
+				MatchErrKeyWords(
+					VerifyFailed,
+					"CRL file cache miss",
+					"Retrieving crl bundle from file cache with key",
+					"Storing crl bundle to file cache with key",
+					"is revoked",
+					"OCSP check failed with unknown error and fallback to CRL check for certificate #2",
+				)
+
+			// verify with cache
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-d").
+				MatchErrKeyWords(
+					VerifyFailed,
+					"Retrieving crl bundle from file cache with key",
+					"is revoked",
+					"OCSP check failed with unknown error and fallback to CRL check for certificate #2",
+				).
+				NoMatchErrKeyWords(
+					"CRL file cache miss",
+					"Storing crl bundle to file cache with key",
+				)
+		})
+	})
+
+	It("successfully completed with cache creation error in debug log", func() {
```

Comment:
You may want to rename this as it's not in the debug log anymore.

Reply:
Updated.